src/crystal/compiler_rt.cr

```diff
@@ -9,3 +9,70 @@ require "./compiler_rt/divmod128.cr"
   # __multi3 was only missing on arm
   require "./compiler_rt/multi3.cr"
 {% end %}
+
+{% if flag?(:win32) && flag?(:bits64) %}
```

Comment:
I think these defs should go into `primitives.cr`, not `compiler_rt.cr`. It's confusing to have definitions there but they're actually overridden in some other place.

Reply:
This would force a dependency from `primitives.cr` on compiler-rt and, as I explained, the compiler-rt functions are not ready because the empty prelude requires `primitives.cr` and those functions need parts of the standard library.

If we really want to not define them in `primitives.cr` at all, that is doable too:

```crystal
{% begin %}
  {% ints = %w(Int8 Int16 Int32 Int64 Int128 UInt8 UInt16 UInt32 UInt64 UInt128) %}
  # ...

  {% for int in ints %}
    struct {{int.id}}
      {% for int2 in ints %}
        # don't define `#unsafe_div` and `#unsafe_mod` here
      {% end %}
      # ...
    end
  {% end %}

  # ...

  {% skip_bad_128bit_primitives = flag?(:win32) && flag?(:bits64) && !flag?(:skip_crystal_compiler_rt) %}
  {% ints = %w(Int8 Int16 Int32 Int64 UInt8 UInt16 UInt32 UInt64) if skip_bad_128bit_primitives %}

  {% for int in ints %}
    struct {{int.id}}
      {% for int2 in ints %}
        @[Primitive(:binary)]
        def unsafe_div(other : {{int2.id}}) : self
        end

        @[Primitive(:binary)]
        def unsafe_mod(other : {{int2.id}}) : self
        end
      {% end %}
      # ...
    end
  {% end %}
{% end %}
```

A consequence is that those divisions and modulos will straight up become compilation errors with an empty prelude because they are truly undefined.

Reply:
I think it's fine that these me functions error at compile time when they're not defined. It only happens when they're actually called. And I think that's better than being silent and just fail at runtime.